inactive_session_timeout/models/res_users.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+
+#     This file is part of inactive_session_timeout, an Odoo module.
+#
+#     Copyright (c) 2015 ACSONE SA/NV (<http://acsone.eu>)
+#
+#     inactive_session_timeout is free software: you can redistribute it
+#     and/or modify it under the terms of the GNU Affero General Public License
+#     as published by the Free Software Foundation, either version 3 of
+#     the License, or (at your option) any later version.
+#
+#     inactive_session_timeout is distributed in the hope that it will
+#     be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the
+#     GNU Affero General Public License
+#     along with inactive_session_timeout.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models
+from openerp import http
+from openerp.http import root
+from openerp.http import request
+from openerp.addons.inactive_session_timeout.models.model import \
+    InactiveSessionTimeoutInstalled
+
+from os import utime
+from os.path import getmtime
+from time import time
+
+DELAY_KEY = 'inactive_session_time_out_delay'
+IGNORED_PATH_KEY = 'inactive_session_time_out_ignored_url'
+
+
+class res_users(models.Model):
+    _inherit = 'res.users'
+
+    def _check_session_validity(self, db, uid, passwd):
+        if not request:
+            return
+        session = request.session
+        session_store = root.session_store
+        registry = self.pool
+        if InactiveSessionTimeoutInstalled._name not in registry:
+            # module not installed in database,
+            # continue usual behavior
+            return
+        param_model = registry['ir.config_parameter']
+        cr = self.pool.cursor()
+        try:
+            delay = int(param_model.get_param(cr, uid, DELAY_KEY, 7200))
+            urls = param_model.get_param(cr, uid, IGNORED_PATH_KEY,
+                                         '').split(',')
```

Comment:
this causes 2 extra db requests per http requests, shouldn't you better cache this values?